test/system/010-images.bats

```diff
@@ -378,16 +378,24 @@ EOF
 
     # IMPORTANT! Use -2/-1 indices, not 0/1, because $SYSTEMD_IMAGE may be
     # present in store, and if it is it will precede $IMAGE.
-    CONTAINERS_STORAGE_CONF=$sconf run_podman images -a -n --format "{{.Repository}}:{{.Tag}} {{.ReadOnly}}"
+    CONTAINERS_STORAGE_CONF=$sconf run_podman images -a -n --format "{{.ID}} {{.Repository}}:{{.Tag}} {{.ReadOnly}}"
     assert "${#lines[*]}" -ge 2 "at least 2 lines from 'podman images'"
-    is "${lines[-2]}" "$IMAGE false" "image from readonly store"
-    is "${lines[-1]}" "$IMAGE true" "image from readwrite store"
-
-    CONTAINERS_STORAGE_CONF=$sconf run_podman images -a -n --format "{{.Id}}"
-    id=${lines[-1]}
+    assert "${lines[-2]}" =~ ".*$IMAGE false" "image from readwrite store"
+    assert "${lines[-1]}" =~ ".*$IMAGE true" "image from readonly store"
+    id=${lines[-2]%% *}
+    local cd; cd=$(image_config_digest "@$id") # Without $sconf, i.e. from the read-write store.
 
     CONTAINERS_STORAGE_CONF=$sconf run_podman pull -q $IMAGE
-    is "$output" "$id" "pull -q $IMAGE, using storage.conf"
+    # This is originally a regression test, (podman pull) used to output multiple image IDs. Ensure it only prints one.
+    [[ $(printf '%s' "$output" | wc -l) -le 1 ]]
```

Comment:
Suggestion:
```bash
    assert "${#lines[*]}" -le 1 "Number of output lines from podman pull"
```
Reason: the Bats `[[ etc ]]` tests generate error diagnostics that are ... sub-ideal. This is an assertion that must never fail, but even those (especially those?) deserve a way to help someone debug.

Reply:
Thanks, I completely forgot `assert` exists despite using it elsewhere.

Fixed as proposed.